src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -546,40 +560,46 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
             activeComponent.updateSubmissionFromView();
         }
 
-        // goes through all exercises and checks if there are unsynced submissions
+        // go through ALL student exam exercises and check if there are unsynced submissions
+        // we do this, because due to connectivity problems, other submissions than the currently active one might have not been saved to the server yet
         const submissionsToSync: { exercise: Exercise; submission: Submission }[] = [];
         this.studentExam.exercises!.forEach((exercise: Exercise) => {
-            exercise.studentParticipations!.forEach((participation) => {
-                participation
-                    .submissions!.filter((submission) => !submission.isSynced)
-                    .forEach((unsynchedSubmission) => {
-                        submissionsToSync.push({ exercise, submission: unsynchedSubmission });
-                    });
-            });
+            if (exercise.studentParticipations) {
+                exercise.studentParticipations!.forEach((participation) => {
+                    if (participation.submissions) {
+                        participation.submissions
+                            .filter((submission) => !submission.isSynced)
+                            .forEach((unsynchedSubmission) => {
+                                submissionsToSync.push({ exercise, submission: unsynchedSubmission });
+                            });
+                    }
+                });
```

Comment:
this makes it less code.

```suggestion
                exercise.studentParticipations!.forEach((participation) => {
                participation
                    .submissions?.filter((submission) => !submission.isSynced)
                    .forEach((unsynchedSubmission) => {
                        submissionsToSync.push({ exercise, submission: unsynchedSubmission });
                    });
            });
```